ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js

```diff
@@ -11,252 +11,252 @@
-                // make sure the current model has width and height before drawing
-                if (polygon && !_.isUndefined(polygon)) {
-                    this.drawBorderedPolygon(polygon);
-                }
-            },
-
-            updateGeometry: function(model) {
-                var rectangle = this.modelToPolygon(model);
-                if (rectangle) {
-                    this.drawBorderedPolygon(rectangle);
-                }
-            },
-
-            drawBorderedPolygon: function(rectangle) {
-
-                if (!rectangle) {
-                    // handles case where model changes to empty vars and we don't want to draw anymore
-                    return;
-                }
-
-                if (this.vectorLayer) {
-                    this.map.removeLayer(this.vectorLayer);
-                }
-
-                this.billboard = new ol.Feature({
-                    geometry: rectangle
-                });
+        // Ensure that the first and last coordinate are the same
+        if (!_.isEqual(coords[0], coords[coords.length - 1])) {
+            coords.push(coords[0]);
+        }
+        return new ol.geom.LineString(coords);
+    },
+
+    modelToPolygon: function(model) {
+        const coords = model.get('polygon');
+        if (!coords) { return; }
+        const isMultiPolygon = ShapeUtils.isArray3D(coords);
+        const multiPolygon = isMultiPolygon ? coords : [coords];
+
+        const multiLineString = new ol.geom.MultiLineString([]);
+        _.each(multiPolygon, function(polygon) {
+            multiLineString.appendLineString(this.coordsToLineString(polygon));
+        }.bind(this));
+
+        return multiLineString;
+    },
+
+    updatePrimitive: function(model) {
+        const polygon = this.modelToPolygon(model);
+        // make sure the current model has width and height before drawing
+        if (polygon && !_.isUndefined(polygon)) {
+            this.drawBorderedPolygon(polygon);
+        }
+    },
+
+    updateGeometry: function(model) {
+        const rectangle = this.modelToPolygon(model);
+        if (rectangle) {
+            this.drawBorderedPolygon(rectangle);
+        }
+    },
+
+    drawBorderedPolygon: function(rectangle) {
+
+        if (!rectangle) {
+            // handles case where model changes to empty vars and we don't want to draw anymore
+            return;
+        }
+
+        if (this.vectorLayer) {
+            this.map.removeLayer(this.vectorLayer);
+        }
+
+        const bufferWidth = DistanceUtils.getDistanceInMeters(this.model.get('polygonBufferWidth'), this.model.get('polygonBufferUnits')) || 1;
+        const line = Turf.lineString(translateFromOpenlayersCoordinates(rectangle.getCoordinates()));
+        const bufferedLine = Turf.buffer(line, bufferWidth, 'meters');
+        const geometryRepresentation = (bufferedLine && new ol.geom.Polygon(bufferedLine.geometry.coordinates)) || rectangle;
```

Comment:
❓ Is the check for `bufferedLine` necessary?

Reply:
Are you asking whether we need this guard at all or whether it should go up a couple of lines into it's own block?